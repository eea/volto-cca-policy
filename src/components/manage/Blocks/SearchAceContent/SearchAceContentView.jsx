import { List, Icon } from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import { flattenToAppURL } from '@plone/volto/helpers';

export default function SearchAceContentView(props) {
  const { data } = props;
  const results = data._v_results || [];

  // console.log(data);

  return results && results.length > 0 ? (
    <div className="block search-acecontent-block">
      {data.title && <h4>{data.title}</h4>}
      <List>
        {results.map((result, index) => (
          <List.Item key={index}>
            <Link to={flattenToAppURL(result[2])}>
              {result[0]} ({result[1]})
            </Link>
          </List.Item>
        ))}
      </List>
      <Link
        to="/en/help/share-your-info"
        className="ui button icon left labeled primary"
      >
        <Icon name="ri-share-line" />
        Share your information
      </Link>
    </div>
  ) : (
    <div>No results</div>
  );
}
