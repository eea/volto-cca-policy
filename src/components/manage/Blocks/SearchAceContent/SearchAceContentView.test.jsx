import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-intl-redux';
import SearchAceContentView from './SearchAceContentView';

const mockStore = configureStore();

jest.mock('semantic-ui-react', () => ({
  ...jest.requireActual('semantic-ui-react'),
  Icon: () => <div>Icon</div>,
}));

describe('SearchAceContentView', () => {
  it('should render the component', () => {
    const data = {
      _v_results: [
        ['Publications and reports', 154, '/', []],
        ['Information portals', 57, '/', []],
      ],
    };

    const store = mockStore({
      userSession: { token: '1234' },
      intl: {
        locale: 'en',
        messages: {},
      },
    });

    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <SearchAceContentView data={data} />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
