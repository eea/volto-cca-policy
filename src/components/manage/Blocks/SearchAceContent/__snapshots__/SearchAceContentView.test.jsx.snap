// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchAceContentView should render the component 1`] = `
<div
  className="block search-acecontent-block"
>
  <div
    className="ui list"
    role="list"
  >
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Publications and reports
         (
        154
        )
      </a>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Information portals
         (
        57
        )
      </a>
    </div>
  </div>
  <a
    className="ui button icon left labeled primary"
    href="/en/help/share-your-info"
    onClick={[Function]}
  >
    <div>
      Icon
    </div>
    Share your information
  </a>
</div>
`;
